module/data/region-behavior/rotate-area.mjs

```diff
@@ -0,0 +1,380 @@
+    }
+  }
+
+  /* ---------------------------------------- */
+  /*  Hooks                                   */
+  /* ---------------------------------------- */
+
+  /**
+   * Stores previous position for animation.
+   * @param {CanvasDocument} doc
+   * @param {object} changes
+   * @param {object} options
+   */
+  static preUpdatePlaceable(doc, changes, options) {
+    if ( foundry.utils.getProperty(options, "dnd5e.animate.moveDelta") && ( ("x" in changes) && ("y" in changes) ) ) {
+      foundry.utils.setProperty(options, `dnd5e.animate.moveOrigin.${doc.id}`, { x: doc.x, y: doc.y });
+    }
+  }
+
+  /* ---------------------------------------- */
+
+  /**
+   * Animate placeable movement.
+   * @param {CanvasDocument} doc
+   * @param {object} changes
+   * @param {object} options
+   */
+  static updatePlaceable(doc, changes, options) {
+    const originalPosition = foundry.utils.getProperty(options, `dnd5e.animate.moveOrigin.${doc.id}`);
+    const angle = foundry.utils.getProperty(options, "dnd5e.animate.angle");
+    const pivot = foundry.utils.getProperty(options, "dnd5e.animate.pivot");
+    if ( !angle ) return;
+    const duration = foundry.utils.getProperty(options, "dnd5e.animate.duration") ?? 500;
+    foundry.canvas.animation.CanvasAnimation.animate([
+      {
+        attribute: "x",
+        parent: doc,
+        from: originalPosition?.x ?? doc.x,
+        to: doc.x
+      },
+      {
+        attribute: "y",
+        parent: doc,
+        from: originalPosition?.y ?? doc.y,
+        to: doc.y
+      },
+      {
+        attribute: "rotation",
+        parent: doc,
+        from: doc.rotation - angle,
+        to: doc.rotation
+      }
+    ], {
+      duration,
+      easing: foundry.canvas.animation.CanvasAnimation.easeInOutCosine,
+      priority: PIXI.UPDATE_PRIORITY.HIGH,
+      ontick: (e, a) => RotateAreaRegionBehaviorType.#animateFrame(doc, angle, pivot, e, a)
+    });
+  }
+
+  /* ---------------------------------------- */
+
+  /**
+   * Handle manually updating animation per-frame.
+   * @param {CanvasDocument} doc
+   * @param {Degrees} angle
+   * @param {Point} pivot
+   * @param {number} elapsedMS               The incremental time in MS which has elapsed (uncapped).
+   * @param {CanvasAnimationData} animation  The animation which is being performed.
+   */
+  static #animateFrame(doc, angle, pivot, elapsedMS, animation) {
+    const pt = animation.time >= animation.duration ? 1 : animation.time / animation.duration;
+
+    if ( (pt !== 1) ) {
+      const pa = animation.easing?.(pt) ?? pt;
+      const pr = Math.toRadians(angle * pa);
+      let from = {};
+      animation.attributes.forEach(a => from[a.attribute] = a.from);
+      const size = RotateAreaRegionBehaviorType.#placeableSize(doc);
+      from.x += size.width / 2;
+      from.y += size.height / 2;
+      Object.assign(doc, RotateAreaRegionBehaviorType.#calculatePosition(pr, pivot, from, size));
+    }
+
+    doc.object.draw();
```

Comment:
Shouldn't need to call `draw`, that's very heavyweight. I would expect `doc.object.renderFlags.set({ refresh: true })` to be enough. Since we know only `x`, `y`, and `rotation` are affected, we can even be more granular:

```suggestion
    doc.object.renderFlags.set({ refreshPosition: true, refreshRotation: true });
```